### PR TITLE
feat: Make trees more distinct

### PR DESCRIPTION
**Pull request by skaldarnar**
I used this tool to play around with the L-systems to figure out good production rules and parameters: http://dkandalov.github.io/lsystem/

Requires https://github.com/Terasology/CoreAssets/pull/5

:evergreen_tree: Updated **Pine Tree**:
image

:deciduous_tree: Updated **Birch Tree**:
image

:fallen_leaf: Updated **Red Tree** (basically just manually tinted foliage):
image

### CODE REVIEW

File: module.txt [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "description": "A basic set of world generators and facets.",
     "dependencies": [
         { "id": "BiomesAPI", "minVersion": "4.0.0" },
-        { "id": "CoreAssets", "minVersion": "2.0.1" }
+        { "id": "CoreAssets", "minVersion": "2.2.0-SNAPSHOT" }
```
Comment (jdrueckert) [inline module.txt:10]: Why "snapshot"?

File: src/main/java/org/terasology/core/world/generator/facetProviders/DefaultTreeProvider.java [hunk 2]
```diff
@@ -15,6 +15,7 @@
  */
 package org.terasology.core.world.generator.facetProviders;
 
+import java.util.Arrays;
```
Comment (jdrueckert) [inline src/main/java/org/terasology/core/world/generator/facetProviders/DefaultTreeProvider.java]: where did that come from?

File: src/main/java/org/terasology/core/world/generator/trees/RecursiveTreeGeneratorLSystem.java [hunk 3]
```diff
@@ -35,45 +37,75 @@
...
                     if (depth > 1) {
                         int size = 1;
-
+                        // placing a "sphere" around 'p' with r=~1.5?
+                        //
+                        // □ □ ■ □ □
+                        // □ ■ ■ ■ □
+                        // ■ ■ ⛝ ■ ■
+                        // □ ■ ■ ■ □
+                        // □ □ ■ □ □
```
Comment (jdrueckert) [inline src/main/java/org/terasology/core/world/generator/trees/RecursiveTreeGeneratorLSystem.java:96]: Consider adding such a schema for the cylinder above as well

File: src/main/java/org/terasology/core/world/generator/trees/TreeGeneratorCactus.java [hunk 4]
```diff
@@ -22,15 +22,16 @@
...
 public class TreeGeneratorCactus extends AbstractTreeGenerator {
 
+    private static final int MAX_HEIGHT = 4;
+
     private BlockUri cactusType;
 
     @Override
     public void generate(BlockManager blockManager, CoreChunk view, Random rand, int posX, int posY, int posZ) {
-        for (int y = posY; y < posY + 3; y++) {
+        for (int y = posY; y < posY + rand.nextInt(1, MAX_HEIGHT); y++) {
```
Comment (jdrueckert) [inline src/main/java/org/terasology/core/world/generator/trees/TreeGeneratorCactus.java:34]: I believe the height==1 cacti look a bit lost... maybe consider a min height of 2

File: src/main/java/org/terasology/core/world/generator/trees/RecursiveTreeGeneratorLSystem.java [hunk 5]
```diff
@@ -18,11 +18,13 @@
 
 import java.util.Map;
 
+import com.google.common.primitives.Floats;
```
Comment (jdrueckert) [inline src/main/java/org/terasology/core/world/generator/trees/RecursiveTreeGeneratorLSystem.java]: what is that required for?

File: module.txt (hunk 1 above)
Comment (skaldarnar) [reply module.txt:10]: I don't think we should do a module release for each and every little change. We have the `develop` branches for some reason, so maybe we can use that for "staging" multiple changes before doing a full release.

Thus, I think that `-SNAPSHOT` dependencies are okay on `develop` but have be replaced by releases before merging to `master`...

File: src/main/java/org/terasology/core/world/generator/trees/RecursiveTreeGeneratorLSystem.java (hunk 3 above)
Comment (skaldarnar) [reply src/main/java/org/terasology/core/world/generator/trees/RecursiveTreeGeneratorLSystem.java:96]: Will do when we get to the point where we're actually writing JavaDoc for this :wink:

File: src/main/java/org/terasology/core/world/generator/trees/TreeGeneratorCactus.java (hunk 4 above)
Comment (skaldarnar) [reply src/main/java/org/terasology/core/world/generator/trees/TreeGeneratorCactus.java:34]: Hm, I especially like them :thinking: they remind me of these small, stubby cacti (https://de.wikipedia.org/wiki/Echinocactus_grusonii)

File: module.txt (hunk 1 above)
Comment (jdrueckert) [reply module.txt:10]: While I understand the staging approach, I'm a tad afraid, that this increased maintenance effort will will delay module and as a result also game releases.
But I guess we can do it like that for now and discuss that some place else.

File: src/main/java/org/terasology/core/world/generator/trees/TreeGeneratorCactus.java (hunk 4 above)
Comment (jdrueckert) [reply src/main/java/org/terasology/core/world/generator/trees/TreeGeneratorCactus.java:34]: fair, your PR ;)

**Review by jdrueckert — APPROVED** [review]

**Pull request merged**